coveo-stew/README_MULTIPLE_LIBRARIES.md

```diff
@@ -0,0 +1,117 @@
+
+[tool.stew]
+pydev = true
+
+[build-system]
+requires = ["poetry_core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+```
+
+You can then proceed to generate the `[poetry.tool.dev-dependencies]` section and create your virtual environment:
+
+```
+$ stew pull-dev-requirements
+$ poetry update
+```
+
+The pydev environment has some caveats:
+
+- It cannot be packaged, published or even pip-installed.
+- `stew ci` will skip it.
+- the `tool.poetry.dev-dependencies` section is reserved, can be generated and updated through stew's `pull-dev-requirements` and `fix-outdated` commands.
+
+The reason we generate dev requirements is that `poetry install` will not install `dev-requirements` of dependencies.
+Since all the local projects are dependencies, the `pull-dev-requirements` script will inspect each of them
+in order to aggregate their dev requirements into the pydev environment.
+
+When you add or remove dev-dependencies, run `pull-dev-requirements` again.
+When you change the constraint of a dev-dependency, just run `stew bump` or `poetry update` to lock the new version.
+
+
+# How to depend on a local python library
+
+Python libraries (i.e.: not the pydev one!) may reference other local libraries using a relative path.
+
+This is not particularly well handled by poetry and will cause problems with some operations such as "poetry build". 
+We leverage poetry's `path` constraint in a very specific way to make it work:
+
+```
+[tool.poetry.dependencies]
+my-package = { version = "^2.4" }
+
+[tool.poetry.dev-dependencies]
+my-package = { path = "../my-package/" }
+```
+
+Essentially, the behavior we're looking for:
+
+- Through `pip install`, it will obtain the latest `^2.4` from `pypi.org`
+- Through `poetry install`, which is only meant for development, the source is fetched from the disk
+
+Note: Even if your libraries are not published to a `pypi` server, you can still use `stew build` to create an offline distribution.
+This concept is particularly useful in business scenarios, for instance when dealing with containerized images and cloud storages.
+
+
+# A note about the aggregated environment.
+
+When you develop using the `pydev` environment, you are aggregating the dependencies of all the libraries into one environment.
+This can cause some problems, for instance if your library A and B both require a 3rd party package Z, you must add Z to the `pyproject.toml` of both A and B.
+If you omit Z in B, your pydev environment will still have Z because it was included with A. 
+However, when you publish B, users will most likely run into an ImportError since Z was not specified there.
+
+To circumvent this, `stew ci` mostly skips testing the `pydev` environment.
+It will instead create an environment for A using A's `pyproject.toml` for tests.
+Then it will create the environment for B using B's `pyproject.toml` for tests. 
+And technically, it should explode here if you forgot to include Z!
```

Comment:
Either

```suggestion
Then it will create the environment for B using B's `pyproject.toml` for tests, and technically, it should explode here if you forgot to include Z!
```

or

```suggestion
Then it will create the environment for B using B's `pyproject.toml` for tests.
Technically, it should explode here if you forgot to include Z!
```